### PR TITLE
StageService (for Schlagzeilen on ArticlePages)

### PR DESCRIPTION
**Pull request by harryurban**

### CODE REVIEW

File: stroeer/content/v1/content_service.proto [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+import "stroeer/page/section/v1/section_page.proto";
+
+package stroeer.content.v1;
+
+option go_package = "github.com/stroeer/tapir/go/stroeer/content/v1;content";
+option java_multiple_files = true;
+option java_package = "de.stroeer.content.v1";
+
+service ContentService {
```
Comment (thisismana) [inline stroeer/content/v1/content_service.proto:11]: As our `ArticlePageService` and `SectionPageService`, this should either be a `FragmentService` (or `Partial(s)Service`) because `Content` is too broad.

Another better fitting name would be `StageService`

**Review by thisismana — CHANGES_REQUESTED** [review]

File: stroeer/fragment/stage/v1/stage_service.proto [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+option go_package = "github.com/stroeer/tapir/go/stroeer/fragment/stage/v1;stage";
+option java_multiple_files = true;
+option java_package = "de.stroeer.fragment.stage.v1";
+
+service StageService {
+  /**
+    ############################################################################
+    # Description
+    ############################################################################
+    Get single stages by requesting them via agreed tokens, e.g. "schlagzeilen" or "meistgelesen".
```
Comment (thisismana) [inline stroeer/fragment/stage/v1/stage_service.proto:16]: via well-known tokens?

Comment (harryurban) [reply stroeer/fragment/stage/v1/stage_service.proto:16]: nit-picking?

**Review by thisismana — APPROVED** [review]

**Review by bobaaaaa — APPROVED** [review]

File: stroeer/fragment/stage/v1/stage_service.proto [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    cacheable: no
+  **/
+  rpc GetStages (GetStagesRequest) returns (GetStagesResponse) {
+  }
+}
+
+// Message to request stages independent of SectionPages, for client side rendering or ESIs
+message GetStagesRequest {
+  // an id is a token which is agreed between frontend and backend, for example "schlagzeilen"
+  repeated string ids = 1;
```
Comment (moritzzimmer) [inline stroeer/fragment/stage/v1/stage_service.proto:50]: Did you consider enumerations here? Feels like this is a finite list of possible Ids which could be selected in a section tool.

File: stroeer/fragment/stage/v1/stage_service.proto (hunk 2 above)
Comment (moritzzimmer) [inline stroeer/fragment/stage/v1/stage_service.proto:16]: I find the mix of calling the identifiers `id` and/or `token` in the comments confusing. I'd go with id/identifier

File: stroeer/fragment/stage/v1/stage_service.proto [hunk 4]
```diff
@@ -0,0 +1,58 @@
+syntax = "proto3";
+
+import "stroeer/page/section/v1/section_page.proto";
+
+package stroeer.fragment.stage.v1;
```
Comment (moritzzimmer) [inline stroeer/fragment/stage/v1/stage_service.proto:5]: Late to the naming party: why `fragment.stage` package? Stages are usually part of a section and fragment seems unrelated for me. Are companions also stages?

File: Makefile [hunk 5]
```diff
@@ -168,3 +168,7 @@ release: clean check check-git-branch ## Releases new version of gRPC source cod
 	git push origin $(NEXT_TAG)
 	git tag -a $(NEXT_GO_TAG) -m "$(NEXT_GO_TAG)"
 	git push origin $(NEXT_GO_TAG)
+
+
+release-local-java:
```
Comment (moritzzimmer) [inline Makefile:173]: please add `PHONY` and comment for the help target

File: stroeer/fragment/stage/v1/stage_service.proto [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+option go_package = "github.com/stroeer/tapir/go/stroeer/fragment/stage/v1;stage";
```
Comment (moritzzimmer) [inline stroeer/fragment/stage/v1/stage_service.proto:7]: please add generated go code to the PR

**Review by moritzzimmer — CHANGES_REQUESTED** [review]

File: stroeer/fragment/stage/v1/stage_service.proto (hunk 4 above)
Comment (thisismana) [reply stroeer/fragment/stage/v1/stage_service.proto:5]: not quite correct, we need this very StageService within the ArticleDetailPage. Its neither `core` nor a `page`, that's why I suggested a `fragment` package. Not sure if we really need the `stage` package.

File: stroeer/fragment/stage/v1/stage_service.proto (hunk 3 above)
Comment (thisismana) [reply stroeer/fragment/stage/v1/stage_service.proto:50]: This may be a big enum and change quite often, as there are not only ld-lists in it, but also synthethical stuff like `most-read` and such.
We'd have to update the FE, BE and SectionTool every time something will be added :/

File: stroeer/fragment/stage/v1/stage_service.proto (hunk 4 above)
Comment (bobaaaaa) [reply stroeer/fragment/stage/v1/stage_service.proto:5]: A companion is just a name for a 'area' of content. This content could be (multiple) `stages`, `commercials` and `teaser`

File: stroeer/fragment/stage/v1/stage_service.proto (hunk 3 above)
Comment (harryurban) [reply stroeer/fragment/stage/v1/stage_service.proto:50]: Das was MaNa sagt.

File: stroeer/fragment/stage/v1/stage_service.proto (hunk 2 above)
Comment (harryurban) [reply stroeer/fragment/stage/v1/stage_service.proto]: done

File: Makefile (hunk 5 above)
Comment (harryurban) [reply Makefile]: done

File: stroeer/fragment/stage/v1/stage_service.proto (hunk 6 above)
Comment (harryurban) [reply stroeer/fragment/stage/v1/stage_service.proto]: done and https://github.com/stroeer/tapir/issues/229

File: stroeer/fragment/stage/v1/stage_service.proto (hunk 2 above)
Comment (harryurban) [reply stroeer/fragment/stage/v1/stage_service.proto]: done

File: stroeer/fragment/stage/v1/stage_service.proto (hunk 4 above)
Comment (moritzzimmer) [reply stroeer/fragment/stage/v1/stage_service.proto]: guess I would skip `stage` in the package path, but fine for me

**Review by moritzzimmer — APPROVED** [review]
